src/actions.rs

```diff
@@ -0,0 +1,385 @@
+use std::{collections::HashMap, io::SeekFrom};
+
```

Comment:
Please create one commit moving code around between files and a *seperate* commit wit the logging stuff